src/error.rs

```diff
@@ -135,6 +153,35 @@ impl From<serde_json::Error> for Error {
     }
 }
 
+impl From<bs58::decode::Error> for Error {
+    fn from(err: bs58::decode::Error) -> Self {
+        match err {
+            bs58::decode::Error::BufferTooSmall => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            bs58::decode::Error::InvalidCharacter {
```

Comment:
I believe `InvalidCharacter` and `NonAsciiCharacter` are quite expected errors during decoding meaning that the message is Malformed. So, I think it should generate Malformed error kind, not InvalidState.

Reply:
fixed